source/directives/text.ts

```diff
@@ -0,0 +1,48 @@
+/// <reference path="../../node_modules/forms-js/dist/forms-js.d.ts" />
+
+module adaptor.directives {
+
+  export function TextDirective($log:ng.ILogService):ng.IDirective {
+
+    return {
+      require: '^fjsForm',
+      restrict: 'EA',
+      templateUrl: '/templates/text.html',
+
+      scope: {
+        fieldName: '@'
+      },
+
+      link: function($scope:any,
+                     $element:ng.IAugmentedJQuery,
+                     $attributes:ng.IAttributes,
+                     formForController:any):void {
+
+        if (!$scope.fieldName) {
+          $log.error('Missing required field "fieldName"');
+
+          return;
+        }
+
+        $scope.attributeMetadata = formForController.form.registerAttribute($scope.fieldName);
+        $scope.watchPath = 'attributeMetadata.form.formData.' + $scope.fieldName;
+
+        // TODO This is likely a race condition.
+        // TODO Refactor this to be shared between input directives.
+        $scope.$watch($scope.watchPath, function(value:any) {
+          $scope.bindable = formsjs.Flatten.read($scope.fieldName, $scope.attributeMetadata.form.formData);
```

Comment:
Now I understand what the "Flatten" thing does a bit more, but how would the fieldname in the html look if you have a deep nested formData structure?